### PR TITLE
Users/alkerezs/implement column descriptors

### PR DESCRIPTION
**Pull request by kkerezsi**
# Pull Request

## 🤨 Rationale

To have a more versatile naming strategy for the columns that allow replacing IDs with actual entity aliases, it is necessary to have a column descriptor defined on the FieldDTO for the Calibration Forecast.

## 👩‍💻 Implementation

- Added column descriptor on the Calibration Forecast Column response model.
- Implemented dynamic naming strategy for the column

## 🧪 Testing

- Unit tests
- Manual tests

## ✅ Checklist

- [ ] This PR has a title that follows the [commit message format](https://github.com/ni/systemlink-grafana-plugins#commit-message-format).

### CODE REVIEW

File: src/datasources/asset-calibration/AssetCalibrationDataSource.test.ts [hunk 1]
```diff
@@ -23,83 +24,83 @@ beforeEach(() => {
...
 {
-    calibrationForecast: {
-        columns: [
-            { name: "Month", values: ["2022-01-01T00:00:00.0000000Z", "2022-02-01T00:00:00.0000000Z", "2022-03-01T00:00:00.0000000Z"] },
-            { name: "Assets", values: [1, 0, 3] },
-        ]
-    }
+  calibrationForecast: {
+    columns: [
+      { name: "", values: ["2022-01-01T00:00:00.0000000Z", "2022-02-01T00:00:00.0000000Z", "2022-03-01T00:00:00.0000000Z"], columnDescriptors: [{ value: "Month", type: ColumnDescriptorType.Time }] },
```
Comment (CiprianAnton) [inline src/datasources/asset-calibration/AssetCalibrationDataSource.test.ts:29]: Is name  of backwards compatibility? Do we need it?

File: src/datasources/asset-calibration/AssetCalibrationDataSource.test.ts [hunk 2]
```diff
@@ -23,83 +24,83 @@ beforeEach(() => {
...
 {
-    calibrationForecast: {
-        columns: [
-          { name: "Model1 - Localtion1", values: [1] },
-          { name: "Model2 - Localtion1", values: [2] },
-        ]
-    }
+  calibrationForecast: {
+    columns: [
+      { name: "", values: [1], columnDescriptors: [{ value: "Model1", type: ColumnDescriptorType.StringValue }, { value: "Localtion1", type: ColumnDescriptorType.StringValue }] },
```
Comment (CiprianAnton) [inline src/datasources/asset-calibration/AssetCalibrationDataSource.test.ts:88]: Minor and optional: typo on Localtion1

File: src/datasources/asset-calibration/types.ts [hunk 3]
```diff
@@ -141,5 +141,21 @@ export enum AssetFilterProperties {
...
+  columns: FieldDTOWithDescriptor[],
+}
+
+export interface FieldDTOWithDescriptor extends FieldDTO {
+  columnDescriptors: ColumnDescriptor[]
+}
+
+export interface ColumnDescriptor {
+  value: string
+  type: string
```
Comment (saradei-ni) [inline src/datasources/asset-calibration/types.ts:153]:
```suggestion
  type: ColumnDescriptorType
```

I think we could use the enum here so it's clear what to expect.

File: src/datasources/asset-calibration/AssetCalibrationDataSource.ts [hunk 4]
```diff
@@ -83,13 +85,19 @@ export class AssetCalibrationDataSource extends DataSourceBase<AssetCalibrationQ
...
+      const columnName = this.createColumnNameFromDrescriptor(result.fields[columnIndex] as FieldDTOWithDescriptor);
+      const columnValue = result.fields[columnIndex].values?.at(0);
+      formattedFields[0].values!.push(columnName);
+      formattedFields[1].values!.push(columnValue);
     }
 
     result.fields = formattedFields;
   }
 
+  createColumnNameFromDrescriptor(field: FieldDTOWithDescriptor): string {
```
Comment (CiprianAnton) [inline src/datasources/asset-calibration/AssetCalibrationDataSource.ts:97]: Typo

File: src/datasources/asset-calibration/types.ts (hunk 3 above)
Comment (CiprianAnton) [inline src/datasources/asset-calibration/types.ts:153]: Should this be ColumnDescriptorType ?

**Review by CiprianAnton — APPROVED** [review]

**Review by saradei-ni — APPROVED** [review]

**Comment by mure** [comment]
@kkerezsi FYI this PR's title did not follow the [format](https://github.com/ni/systemlink-grafana-plugins#commit-message-format) and therefore did not generate a new release

**Comment by kkerezsi** [comment]
@mure I missed this. My bad . Will pay more attention to this on the next PRs

**Pull request merged**